numpy/core/function_base.py

```diff
@@ -82,6 +82,7 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
 
     """
     num = int(num)
+    assert num >= 0
```

Comment:
Should raise a ValueError (with some detailed text). Also needs add a test, there should be tests for linspace already somewhere in /numpy/core/tests/test_function_base.py or so.